src/timeseriesflattener/flattened_dataset.py

```diff
@@ -414,10 +416,17 @@ def _get_temporal_feature(
         Returns:
             pd.DataFrame: Feature
         """
-
-        if self.cache and self.cache.feature_exists(feature_spec=feature_spec):
-            df = self.cache.read_feature(feature_spec=feature_spec)
-            return df.set_index(keys=self.pred_time_uuid_col_name).sort_index()
+        # Create copies of objects to avoid competing threads being IO bound
+        # whent they try to read/write to the same object
```

Comment:
Where is the copying being done? Add a reference to the function that does it, since it doesn't follow in the next lines